kong/plugins/zipkin/opentracing.lua

```diff
@@ -50,57 +53,67 @@ local function ip_tag(addr)
   end
 end
 
+
+-- adds the proxy span to the opentracing context, unless it already exists
+local function get_or_add_proxy_span(opentracing, timestamp)
+  if not opentracing.proxy_span then
+    local request_span = opentracing.request_span
+    opentracing.proxy_span = request_span:start_child_span(
+      request_span.name .. " (proxy)",
+      timestamp)
+    opentracing.proxy_span:set_tag("span.kind", "client")
+  end
+  return opentracing.proxy_span
+end
+
+
 if subsystem == "http" then
   function OpenTracingHandler:initialise_request(conf, ctx)
     local tracer = self:get_tracer(conf)
     local req = kong.request
     local wire_context = tracer:extract("http_headers", req.get_headers()) -- could be nil
-    local method, url
     local path_with_query = req.get_path_with_query()
-    if path_with_query ~= "" then
-      method = req.get_method()
-      url = req.get_scheme() .. "://" .. req.get_host() .. ":"
-        .. req.get_port() .. path_with_query
-    end
+    local method = req.get_method()
+    local url = req.get_scheme() .. "://" .. req.get_host() .. ":"
+             .. req.get_port() .. path_with_query
     local forwarded_ip = kong.client.get_forwarded_ip()
-    local request_span = tracer:start_span("kong.request", {
-      child_of = wire_context;
+
+    local request_span = tracer:start_span(method .. " " .. url, {
```

Comment:
Another one: calling a span name `GET /something/123abc` is a high cardinality source. In http servers we use `GET /something/{id}` but since kong is an api-gateway it is maybe easier to call it just `GET`.

Reply:
Gotcha, will do. Thanks!